deepchem/models/lightning/dc_lightning_module.py

```diff
@@ -0,0 +1,99 @@
+import torch
+
+try:
+  import pytorch_lightning as pl  # noqa
+  PYTORCH_LIGHTNING_IMPORT_FAILED = False
+except ImportError:
+  PYTORCH_LIGHTNING_IMPORT_FAILED = True
+
+
+class DCLightningModule(pl.LightningModule):
+  """DeepChem Lightning Module to be used with Lightning trainer.
+
+  Example code
+  >>> import deepchem as dc
+  >>> from deepchem.models import MultitaskClassifier
+  >>> import numpy as np
+  >>> import torch
+  >>> from torch.utils.data import DataLoader
+  >>> from deepchem.models.lightning.dc_lightning_module
+  ...   import DCLightningModule
+  >>> model = MultitaskClassifier(params)
+  >>> valid_dataloader = DataLoader(test_dataset)
+  >>> lightning_module = DCLightningModule(model)
+  >>> trainer = pl.Trainer(max_epochs=1)
+  >>> trainer.fit(lightning_module, valid_dataloader)
+
+  The lightning module is a wrapper over deepchem's torch model.
+  This module directly works with pytorch lightning trainer
+  which runs training for multiple epochs and also is responsible
+  for setting up and training models on multiple GPUs.
+  """
+
+  def __init__(self, dc_model):
+    """Create a new DCLightningModule
+
+    Parameters
+    ----------
+    dc_model: deepchem.models.torch_models.torch_model.TorchModel
+      TorchModel to be wrapped inside the lightning module.
+    """
+    super().__init__()
+    self.dc_model = dc_model
+
+    self.pt_model = self.dc_model.model
+    self.loss = self.dc_model._loss_fn
+
+  def configure_optimizers(self):
+    """Configure optimizers, for details refer to:
+    https://pytorch-lightning.readthedocs.io/en/stable/api/pytorch_lightning.core.LightningModule.html?highlight=LightningModule
```

Comment:
You should probably add the link to a References section in the main docstring to render it nicely